### PR TITLE
Adding AuthenticodeSign verification task. Adding test for the task

### PR DESCRIPTION
**Pull request by shahabhijeet**
Adding task for verifying if binaries and powershell files are AuthenticodeSigned.
Adding tests for the task.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@shahabhijeet__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (abhishah). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by shahabhijeet** [comment]
Sign in build http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/648/console
Check for the following output in logs
!!!AuthenticodeSignature status is 'Valid' for file

File: tools/BuildPackagesTask/VerifyAuthenticodeSignatureTask.cs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            //}
+
+            return isSigned;
+        }
+
+        /// <summary>
+        /// This function will check validity of Digital Signature
+        /// </summary>
+        /// <returns>True: If Certificate is valid, False:If Certificate is invalid or expired</returns>
+        private bool CheckCertificate()
```
Comment (markcowl) [inline tools/BuildPackagesTask/VerifyAuthenticodeSignatureTask.cs]: Is this code called anywhere?  If not, please remove.

File: build.proj [hunk 2]
```diff
@@ -261,6 +262,10 @@
         ExpectedDelaySigned="false"
         ContinueOnError="false" 
         Condition="!$(DelaySign) and '@(DelaySignedAssembliesToSign)' != ''"/>
+		
+	<VerifyAuthenticodeSignatureTask SignedFiles="@(DelaySignedAssembliesToSign);@(ScriptsToSign)">
```
Comment (markcowl) [inline build.proj]: This is already done in the signing task - what we need is to include the files that we did not build (those files copied by the compiler, usua;lly from nuget packages)

File: tools/BuildPackagesTask/VerifyAuthenticodeSignatureTask.cs (hunk 1 above)
Comment (shahabhijeet) [inline tools/BuildPackagesTask/VerifyAuthenticodeSignatureTask.cs]: Removed unused code

File: build.proj (hunk 2 above)
Comment (shahabhijeet) [inline build.proj]: Added a way to verify authenticode signature on the output folder

**Comment by shahabhijeet** [comment]
Sign in job after addressing PR feedback http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/

File: build.proj (hunk 2 above)
Comment (shahabhijeet) [inline build.proj]: All the feedback was incorporated:
1) Now able to verify all Microsoft.*.dll and System.*.dll from the output folder
2) Remove unused code
3) Latest iteration of sign-in job
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/653/consoleFull

File: build.proj [hunk 3]
```diff
@@ -261,6 +262,11 @@
         ExpectedDelaySigned="false"
         ContinueOnError="false" 
         Condition="!$(DelaySign) and '@(DelaySignedAssembliesToSign)' != ''"/>
+		
+	<VerifyAuthenticodeSignatureTask ProbingDirectory="$(PackageDirectory)\$(Configuration)"
+									FileFilterPattern="microsoft.*.dll;system.*.dll;*.ps1">
```
Comment (markcowl) [inline build.proj]: add *.psm1

**Pull request merged**